__tests__/table_moveColumn.js

```diff
@@ -0,0 +1,476 @@
+    it('keep the merged content columns order', () => {
+      const {
+        state: { tr }
+      } = createEditor(
+        doc(
+          table(
+            row(tdEmpty, td(p('1')), td(p('2'))),
+            row(tdEmpty, td({ colspan: 2 }, p('3'))),
+            row(tdEmpty, td(p('4')), td(p('5')))
+          )
+        )
+      );
+
+      const newTr = moveColumn(1, 0)(tr);
+      expect(newTr.doc).toEqualDocument(
+        doc(
+          table(
+            row(td(p('1')), td(p('2')), tdEmpty),
+            row(td({ colspan: 2 }, p('3')), tdEmpty),
+            row(td(p('4')), td(p('5')), tdEmpty)
+          )
+        )
+      );
+    });
+
+    it('should not move columns when there is not movement', () => {
+      const {
+        state: { tr }
+      } = createEditor(
+        doc(
+          table(
+            row(td({ colspan: 3 }, p('merged cell'))),
+            row(td(p('2')), tdCursor, tdEmpty),
+            row(td(p('3')), tdEmpty, tdEmpty)
+          )
+        )
+      );
+
+      const newTr = moveColumn(1, 0)(tr);
+      expect(newTr.doc).toEqualDocument(
+        doc(
+          table(
+            row(td({ colspan: 3 }, p('merged cell'))),
+            row(td(p('2')), tdCursor, tdEmpty),
+            row(td(p('3')), tdEmpty, tdEmpty)
+          )
+        )
+      );
+    });
+
+    describe('table 3x5', () => {
+      const {
+        state: { tr }
+      } = createEditor(
+        doc(
+          table(
+            row(
+              td({ rowspan: 2 }, p('M1')),
+              td(p('0')),
+              td(p('1')),
+              td(p('2')),
+              td(p('3'))
+            ),
+            row(
+              td({ colspan: 2 }, p('4')),
+              td({ rowspan: 2, colspan: 2 }, p('M2'))
+            ),
+            row(td(p('5')), td(p('6')), td(p('7')))
+          )
+        )
+      );
+
+      const expectedResult = doc(
+        table(
+          row(
+            td({ rowspan: 2 }, p('M1')),
+            td(p('2')),
+            td(p('3')),
+            td(p('0')),
+            td(p('1'))
+          ),
+          row(
+            td({ rowspan: 2, colspan: 2 }, p('M2')),
+            td({ colspan: 2 }, p('4'))
+          ),
+          row(td(p('5')), td(p('6')), td(p('7')))
+        )
+      );
+
+      it('should move column 1 to 3', () => {
+        let newTr = moveColumn(1, 3)(tr);
+        expect(newTr.doc).toEqualDocument(expectedResult);
+      });
+
+      it('should move column 2 to 3', () => {
+        let newTr = moveColumn(2, 3)(tr);
+        expect(newTr.doc).toEqualDocument(expectedResult);
+      });
+
+      it('should move column 2 to 4', () => {
+        let newTr = moveColumn(2, 4)(tr);
+        expect(newTr.doc).toEqualDocument(expectedResult);
+      });
+
+      it('should move column 3 to 1', () => {
+        let newTr = moveColumn(3, 1)(tr);
+        expect(newTr.doc).toEqualDocument(expectedResult);
+      });
+
+      it('should move column 3 to 2', () => {
```

Comment:
The function will use a "smart fit" strategy:  

<img width="858" alt="Screen Shot 2019-03-27 at 6 21 19 pm" src="https://user-images.githubusercontent.com/3452187/55057404-3b925e80-50bd-11e9-9f8a-b66a76758cec.png">

and this is the expected result:

<img width="845" alt="Screen Shot 2019-03-27 at 6 23 34 pm" src="https://user-images.githubusercontent.com/3452187/55057497-75fbfb80-50bd-11e9-9436-e1dac0c854e5.png">

Reply:
My worry is that "smart fit" might be too smart, because users might want to have a control over where to drop the columns (right/left from the merged section). Also, you never know if the intent was to move it there or was it a mistake. Therefore I'd rather throw an error in this case.

Reply:
That's why I think who should deal with it it's the UI side.